### PR TITLE
Data Explorer

### LINKED ISSUE #108: Data Explorer
- [x] allow free SQL querying of competitive/pro matches
- [x] graphical query builder
~store recent queries in local storage history~
- [x] json download button

### PR DESCRIPTION
**Pull request by howardchung**
fixes #108 

* implements query builder feature
* added link to header navbar

### CODE REVIEW

**Comment by dartharaknus** [comment]
I would love to review this but unfortunately I can't until March 15th. 😞

**Comment by howardchung** [comment]
It'll likely be shipped by then but I'll be happy to respond to a retroactive review

File: src/components/Explorer/Explorer.jsx [hunk 1]
```diff
@@ -130,71 +132,54 @@ class Explorer extends React.Component {
...
+      */}
       <div>
+        <ExplorerFormField label={strings.explorer_select} dataSource={selects} builderField="select" builderContext={this} />
+        <ExplorerFormField label={strings.explorer_group_by} dataSource={groups} builderField="group" builderContext={this} />
+        <ExplorerFormField label={strings.explorer_hero} dataSource={heroes} builderField="hero" builderContext={this} />
+        <ExplorerFormField label={strings.explorer_patch} dataSource={patches} builderField="patch" builderContext={this} />
+        <ExplorerFormField label={strings.explorer_player} dataSource={proPlayers} builderField="player" builderContext={this} />
+        <ExplorerFormField label={strings.explorer_league} dataSource={leagues} builderField="league" builderContext={this} />
+        <ExplorerFormField label={strings.explorer_item} dataSource={items} builderField="item" builderContext={this} />
+        <ExplorerFormField label={strings.explorer_duration} dataSource={durations} builderField="duration" builderContext={this} />
```
Comment (blukai) [inline src/components/Explorer/Explorer.jsx]: Probably here should be one more field `Limit`

File: src/components/Explorer/Explorer.jsx [hunk 2]
```diff
@@ -130,71 +132,54 @@ class Explorer extends React.Component {
...
   render() {
+    const proPlayers = this.props.proPlayers.map(player => ({ text: player.name, value: player.account_id }));
+    const leagues = this.props.leagues.map(league => ({ text: league.name, value: league.leagueid }));
     return (<div>
       <Helmet title={strings.title_explorer} />
-      <Heading title="Data Explorer" subtitle="Explore data from Dota 2 matches" />
-      <ul>
-        <li><a href="https://github.com/odota/core/blob/master/sql/create_tables.sql">Table Schema</a></li>
-        <li>matches and player_matches tables only contain competitive matches in Professional and Premium tiers</li>
-        <li>Queries are read-only and time out after 30 seconds</li>
```
Comment (blukai) [inline src/components/Explorer/Explorer.jsx]: Why delete information about schema and timeout?

File: src/components/Explorer/queryTemplate.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+${patch ? `AND patch = '${patch.value}'` : ''}
+${hero ? `AND hero_id = ${hero.value}` : ''}
+${player ? `AND account_id = ${player.value}` : ''}
+${league ? `AND leagueid = ${league.value}` : ''}
+${item ? `AND (player_matches.purchase->>'${item.value}')::int > 0` : ''}
+${duration ? `AND duration > ${duration.value}` : ''}
+${group ? `GROUP BY ${group.value}` : ''}
+${group ? `HAVING count(${select.value}) > 1` : ''}
+ORDER BY ${group ? 'avg' : select && select.value} DESC NULLS LAST
+LIMIT 500`;
```
Comment (blukai) [inline src/components/Explorer/queryTemplate.js]: Maybe it should be not 500 by default, 100 is enough I guess. Who know, can set limit by yourself.

File: src/components/Explorer/Explorer.jsx (hunk 2 above)
Comment (howardchung) [inline src/components/Explorer/Explorer.jsx]: I thought it looked bad visually, and most users don't need to know it.

File: src/components/Explorer/queryTemplate.js (hunk 3 above)
Comment (howardchung) [inline src/components/Explorer/queryTemplate.js]: I figure users who want to adjust the default limit can just edit the sql query directly themselves.

**Pull request merged**